src/erl.rs

```diff
@@ -222,10 +305,33 @@ fn statement(statement: &TypedStatement, module: &[String]) -> Option<Document>
     }
 }
 
-fn mod_fun(name: &str, args: &[TypedArg], body: &TypedExpr, module: &[String]) -> Document {
+fn mod_fun(
+    name: &str,
+    args: &[TypedArg],
+    body: &TypedExpr,
+    module: &[String],
+    return_type: &Arc<Type>,
+) -> Document {
     let mut env = Env::new(module);
+    // Type Spec
+    // pub type TypedArg = Arg<Arc<Type>>;
+    let return_spec = (&**return_type).to_erlang_type_spec();
+    let args_spec = concat(
+        args.iter()
+            .map(|a| (a.typ).to_erlang_type_spec())
+            .intersperse(", ".to_doc()),
+    );
+    let spec = "-spec "
+        .to_doc()
+        .append(name)
+        .append("(")
+        .append(args_spec)
+        .append(") -> ")
+        .append(return_spec)
+        .append(".")
+        .append(line());
```

Comment:
Would be nice to wrap and intent this when long but I dont' mind as much as the custom type definitions.

Reply:
gave it my best go, passing `args_spec` through `wrap_args` and `return_spec` is going through `to_erlang_type_spec`?  I'm not 100% sure what I'm doing here. :)